src/mpid/common/shm/mpidu_init_shm.c

```diff
@@ -60,7 +59,6 @@ static int Init_shm_barrier()
     if (MPL_atomic_fetch_add_int(&barrier->val, 1) == local_size - 1) {
         MPL_atomic_store_int(&barrier->val, 0);
         MPL_atomic_store_int(&barrier->wait, 1 - sense);
-        MPL_atomic_write_barrier();
```

Comment:
@hzhou just for my better understanding, shouldn't be the write barrier still needed to broadcast changes to other processes?

Reply:
The old code use read/write barrier a lot because it is not using proper atomic variables, often relying on `volatile`. By using more precise atomic operations, it enforces these barrier already. For example, `MPL_atomic_store_int` is an aliias to `MPL_atomic_release_store_int`, which is essentially a relaxed store plus a write flush (barrier).

Reply:
Got it, thanks for the clarification!